packages/react/src/components/AccountSettings/DeleteUser/DeleteUser.tsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+import { Logger } from 'aws-amplify';
+import { deleteUser, translate } from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { Flex } from '../../../primitives';
+import {
+  DefaultWarning,
+  DefaultError,
+  DefaultSubmitButton,
+} from './defaultComponents';
+import { DeleteUserProps, DeleteUserState } from './types';
+
+const logger = new Logger('DeleteUser');
+
+function DeleteUser({
+  onSuccess,
+  onError,
+  handleDelete,
+}: DeleteUserProps): JSX.Element | null {
+  const [state, setState] = React.useState<DeleteUserState>('IDLE');
+  const [errorMessage, setErrorMessage] = React.useState<string>(null);
+
+  // translations
+  const deleteAccountText = translate('Delete Account');
+
+  const { user, isLoading } = useAuth();
+
+  const startConfirmation = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setState('IS_CONFIRMING');
+  };
+
+  const runDeleteUser = React.useCallback(async () => {
+    setState('IN_PROGRESS');
+    try {
+      if (handleDelete) {
+        await handleDelete(user);
+      } else {
+        await deleteUser();
+      }
+      setState('DONE');
+      onSuccess?.();
+    } catch (e) {
+      const error = e as Error;
+      setState('IDLE');
+      setErrorMessage(error.message);
+      onError?.(error);
+    }
+  }, [handleDelete, onError, onSuccess, user]);
```

Comment:
I pulled this to a separate async function, because the event handler for `onConfirmDelete` expects a synchronous function.

Advices to clean this up definitely appreciated.